host/xtest/regression_4000.c

```diff
@@ -6192,3 +6192,498 @@ static void xtest_tee_test_4016_ed25519(ADBG_Case_t *c)
+			continue;
+		ret = process_text_4017(c, &bs, n, middle_count,
+					middle_increment);
+		if (!ADBG_EXPECT_TRUE(c, ret)) {
+			Do_ADBG_Log("Failed processing with initial_count %zu (previous %zu)",
+				n, prev_n);
+			goto out;
+		}
+		prev_n = n;
+	}
+
+	ret = true;
+out:
+	if (bs.oph)
+		ta_crypt_cmd_free_operation(c, s, bs.oph);
+	free(bs.plain_text);
+	return ret;
+}
+
+static void xtest_tee_test_4017(ADBG_Case_t *c)
+{
+	TEEC_Result res = TEEC_SUCCESS;
+	uint32_t ret_orig = 0;
+	TEEC_Session sess = { };
+
+	res = xtest_teec_open_session(&sess, &crypt_user_ta_uuid, NULL,
+				      &ret_orig);
+	if (!ADBG_EXPECT_TEEC_SUCCESS(c, res))
+		return;
+
+	Do_ADBG_BeginSubCase(c, "TEE_ALG_AES_ECB_NOPAD");
+	if (!ADBG_EXPECT_TRUE(c, do_algo_4017(c, &sess, TEE_ALG_AES_ECB_NOPAD,
+					      0)))
+		goto out;
+	Do_ADBG_EndSubCase(c, "TEE_ALG_AES_ECB_NOPAD");
+
+	Do_ADBG_BeginSubCase(c, "TEE_ALG_AES_CBC_NOPAD");
+	if (!ADBG_EXPECT_TRUE(c, do_algo_4017(c, &sess, TEE_ALG_AES_CBC_NOPAD,
+					      0)))
+		goto out;
+	Do_ADBG_EndSubCase(c, "TEE_ALG_AES_CBC_NOPAD");
+
+	Do_ADBG_BeginSubCase(c, "TEE_ALG_AES_CTR");
+	if (!ADBG_EXPECT_TRUE(c, do_algo_4017(c, &sess, TEE_ALG_AES_CTR, 0)))
+		goto out;
+	Do_ADBG_EndSubCase(c, "TEE_ALG_AES_CTR");
+
+	Do_ADBG_BeginSubCase(c, "TEE_ALG_AES_CTR 1 extra byte");
+	if (!ADBG_EXPECT_TRUE(c, do_algo_4017(c, &sess, TEE_ALG_AES_CTR, 1)))
+		goto out;
+	Do_ADBG_EndSubCase(c, "TEE_ALG_AES_CTR 1 extra byte");
+
+	Do_ADBG_BeginSubCase(c, "TEE_ALG_AES_CTS");
+	if (!ADBG_EXPECT_TRUE(c, do_algo_4017(c, &sess, TEE_ALG_AES_CTS, 0)))
+		goto out;
+	Do_ADBG_EndSubCase(c, "TEE_ALG_AES_CTS");
+
+	Do_ADBG_BeginSubCase(c, "TEE_ALG_AES_CTS 1 extra byte");
+	if (!ADBG_EXPECT_TRUE(c, do_algo_4017(c, &sess, TEE_ALG_AES_CTS, 1)))
+		goto out;
+	Do_ADBG_EndSubCase(c, "TEE_ALG_AES_CTS 1 extra byte");
+
+	Do_ADBG_BeginSubCase(c, "TEE_ALG_AES_XTS");
+	if (!ADBG_EXPECT_TRUE(c, do_algo_4017(c, &sess, TEE_ALG_AES_XTS, 0)))
+		goto out;
+	Do_ADBG_EndSubCase(c, "TEE_ALG_AES_XTS");
+
+	Do_ADBG_BeginSubCase(c, "TEE_ALG_AES_XTS 1 extra byte");
+	if (!ADBG_EXPECT_TRUE(c, do_algo_4017(c, &sess, TEE_ALG_AES_XTS, 1)))
+		goto out;
+	Do_ADBG_EndSubCase(c, "TEE_ALG_AES_XTS 1 extra byte");
+
+	Do_ADBG_BeginSubCase(c, "TEE_ALG_AES_GCM");
+	if (!ADBG_EXPECT_TRUE(c, do_algo_4017(c, &sess, TEE_ALG_AES_GCM, 0)))
+		goto out;
+	Do_ADBG_EndSubCase(c, "TEE_ALG_AES_GCM");
+
+	Do_ADBG_BeginSubCase(c, "TEE_ALG_AES_GCM 1 extra byte");
+	if (!ADBG_EXPECT_TRUE(c, do_algo_4017(c, &sess, TEE_ALG_AES_GCM, 1)))
+		goto out;
+	Do_ADBG_EndSubCase(c, "TEE_ALG_AES_GCM 1 extra byte");
+
+	Do_ADBG_BeginSubCase(c, "TEE_ALG_AES_CCM");
+	if (!ADBG_EXPECT_TRUE(c, do_algo_4017(c, &sess, TEE_ALG_AES_CCM, 0)))
+		goto out;
+	Do_ADBG_EndSubCase(c, "TEE_ALG_AES_CCM");
+
+	Do_ADBG_BeginSubCase(c, "TEE_ALG_AES_CCM 1 extra byte");
+	if (!ADBG_EXPECT_TRUE(c, do_algo_4017(c, &sess, TEE_ALG_AES_CCM, 1)))
+		goto out;
+	Do_ADBG_EndSubCase(c, "TEE_ALG_AES_CCM 1 extra byte");
+
+out:
```

Comment:
misses an call to `Do_ADBG_EndSubCase()` when we reach this point before line 6682 is reached.

Actually, I wonder if the `goto out` are really useful. If removed, we'll get a status of all the cases, maybe flowed of error trace messages if many test cases fail but i think its worth it, unless the session crashed soemwhere during the test cases.

Reply:
I'll remove the gotos.